### PR TITLE
Perist selected users in invite modal

### PR DESCRIPTION
**Pull request by t-rutten**
when the user query changes.

### CODE REVIEW

File: app/components/menus/SearchUsersDropDown.jsx [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
-    this.setState({ selectedUsers: [] });
+    try {
+      const selectedUsersIDs = this.state.selectedUsers.map(u => u.id);
+      await this.props.inviteUsers(this.props.postId, selectedUsersIDs);
+      this.setState({ selectedUsers: [] });
+      this.props.clearSearch();
+      this.props.closeModal();
+    }
+    catch (err) {
+      console.error("Sorry, we couldn't invite those users.");
```
Comment (tgoodwin) [inline app/components/menus/SearchUsersDropDown.jsx]: we don't have a solid error reporting pattern yet so tbh this is fine for now but I might add a TODO so we can more easily remember this and do it properly once we're in a better place.

File: app/components/menus/SearchUsersDropDown.jsx [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
+
+  handleCloseModal = () => {
     this.props.clearSearch();
-    this.props.handleClose();
+    this.props.closeModal();
   }
 
   handleSelectInvite(user) {
     const { selectedUsers } = this.state;
-    const idx = selectedUsers.findIndex(uid => uid === user.id);
```
Comment (tgoodwin) [inline app/components/menus/SearchUsersDropDown.jsx]: I may be interpreting this wrong but if you're using `idx` from findIndex just to evaluate if its -1 or not, we could use something more declarative like `const inList = selectedUsers.includes(user.id); if (inList) { do stuff..`

File: app/components/menus/SearchUsersDropDown.jsx [hunk 3]
```diff
@@ -1,24 +1,35 @@
...
+    const idx = selectedUsers.findIndex(u => u.id === user.id);
     if (idx === -1) {
-      this.setState({ selectedUsers: [...selectedUsers, user.id] });
+      this.setState({ selectedUsers: [...selectedUsers, user] });
     } else {
       this.setState({
         selectedUsers: [...selectedUsers.slice(0, idx), ...selectedUsers.slice(idx + 1)]
```
Comment (tgoodwin) [inline app/components/menus/SearchUsersDropDown.jsx]: similarly if we're just trying to drop the value at position `idx` we could use `selectedUsers.filter(uid => uid !== user.id)`

File: app/components/menus/SearchUsersDropDown.jsx [hunk 4]
```diff
@@ -27,20 +38,36 @@ class SearchUsersDropdown extends React.Component {
   }
 
   render() {
+    const { selectedUsers } = this.state;
     const { userResults } = this.props;
-    const userList = userResults ? userResults.map(user => (
+    const userList = userResults ? userResults
+      .filter(user => !this.state.selectedUsers.some(u => u.id === user.id))
+      .map(user => (
```
Comment (tgoodwin) [inline app/components/menus/SearchUsersDropDown.jsx]: damn do you use a functional language at work or something?? 🔥

File: app/components/menus/SearchUsersDropDown.jsx (hunk 1 above)
Comment (t-rutten) [reply app/components/menus/SearchUsersDropDown.jsx]: 👍

File: app/components/menus/SearchUsersDropDown.jsx (hunk 2 above)
Comment (t-rutten) [reply app/components/menus/SearchUsersDropDown.jsx]: you interpreted correctly and you're right, this select/de-select logic could be more clear.

**Pull request merged**